core/process.py

```diff
@@ -41,6 +41,15 @@ def process(reddit, submission, last_switcharoo, action):
         action.act(submission)
         return
 
+    # Verify that the linked post is not NSFW
```

Comment:
This should moved to under the check `# Deleted comments sometimes don't generate errors`. At that point in the flow, it's validated the comment exists and now has a reference to it and the submission so right there would be the best time to check.

Also, I double-checked the rules and actually, individual NSFW submissions are fine but submissions from NSFW subreddits are not. So the check should be something like `comment.subreddit.over_18`. The issue should also probably be renamed to `subreddit_is_NSFW`. Sorry if I miscommunicated that earlier, my bad.